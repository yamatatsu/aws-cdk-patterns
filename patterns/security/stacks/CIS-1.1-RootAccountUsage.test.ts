import { SynthUtils } from "@aws-cdk/assert"
import * as cdk from "@aws-cdk/core"
import * as logs from "@aws-cdk/aws-logs"
import { RootAccountUsage } from "./CIS-1.1-RootAccountUsage"

test("snapshot test", () => {
  const app = new cdk.App()
  const stack = new cdk.Stack(app, "testStack")

  const logGroup = new logs.LogGroup(stack, "test-LogGroup")

  const target = new RootAccountUsage(app, "Target", {
    logGroup,
  })

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
