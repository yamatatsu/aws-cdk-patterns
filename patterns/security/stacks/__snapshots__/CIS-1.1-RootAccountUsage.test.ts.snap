// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "Alarm7103F465": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "TopicBFC7AF6E",
          },
        ],
        "AlarmDescription": "https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cis-controls.html#securityhub-standards-cis-controls-1.1",
        "AlarmName": "CIS-1.1-RootAccountUsage",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "UsedRootAccount",
        "Namespace": "LogMetrics",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 1,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "MetricFilter1B93B6E5": Object {
      "Properties": Object {
        "FilterPattern": "{$.userIdentity.type=\\"Root\\" && $.userIdentity.invokedBy NOT EXISTS && $.eventType !=\\"AwsServiceEvent\\"}",
        "LogGroupName": Object {
          "Fn::ImportValue": "testStack:ExportsOutputReftestLogGroupC4BB93F49A987202",
        },
        "MetricTransformations": Array [
          Object {
            "MetricName": "UsedRootAccount",
            "MetricNamespace": "LogMetrics",
            "MetricValue": "1",
          },
        ],
      },
      "Type": "AWS::Logs::MetricFilter",
    },
    "TopicBFC7AF6E": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
